src/ray/raylet/scheduling/cluster_resource_scheduler.cc

```diff
@@ -248,7 +248,7 @@ int64_t ClusterResourceScheduler::GetBestSchedulableNode(
 
   auto spread_threshold = spread_threshold_;
   // If the scheduling decision is made by gcs, we ignore the spread threshold
-  if (actor_creation && RayConfig::instance().gcs_actor_scheduling_enabled()) {
+  if (grant_or_reject) {
     spread_threshold = 1.0;
   }
```

Comment:
Can we RAY_CHECK later that best_node_id is local node if grant_or_reject is true.

Consider just immediately returning the local node id if feasible here instead of going through the full policy logic.